daml/ContingentClaims/Lifecycle.daml

```diff
@@ -150,41 +148,39 @@ exercise spot election claim today =
   apoCataM pruneZeros' acquireThenExercise
   . (True, ) -- initial election authorizer (`True = bearer`)
   . (MinusInfinity, ) -- initial acquisition time
-  $ claim 
+  $ claim
   where
-    acquireThenExercise = fmap (fmap (fmap join))
-                        . fmap (fmap distE)
-                        . fmap (fmap (fmap (exercise' election today)))
-                        . fmap (fmap sequence)
-                        . fmap sequence . fmap
-                        $ acquire' spot today
-    distE = distApo
+    acquireThenExercise =
+      fmap (exercise' election today)
+      . sequence
+      . fmap (sequence . (Prelude.fst &&& acquire' spot today))
 
--- | Coalgebra for `exercise`.
--- Resolves `Or` nodes by removing them (keeping elected subtrees).
--- Fixes acquisition time of exercised `Anytime` nodes by replacing them with `When (TimeGte t)`.
+-- Resolve `Or` nodes by removing them (keeping elected subtrees).
+-- Fix acquisition time of exercised `Anytime` nodes by replacing them with `When (TimeGte t)`.
 -- The election consists of a boolean representing the authorizer (`True = bearer`),
 -- and this is compared against available branches of the choice.
 exercise' : (Eq t, Eq a)
   => (Bool, C t a)
   -- ^ the election being made
   -> t
   -- ^ the election date
-  -> (Bool, (AcquisitionTime t, C t a))
-  -- ^ the input claim, its sub-trees acquisition time and a flag keeping track of who is the entitled to the election (`True = bearer`)
+  -> (Bool, (AcquisitionTime t, AF t a))
```

Comment:
I'm concerned here that we're moving further from a recursion schemes based design. There, all co-algebras, if I recall correctly are of the form `a -> f (m a)` (or `Functor f , Monad n, Monad m => a -> n (f (m a))` for the monadic case). And at some point, when `AcquisitionTime` was added, this has been lost.

The problem is that we lose composability.

It also makes it much more difficult for somebody familiar with recursion-schemes, but not necessarily our library, to pick it up. This is really a comment for a different PR, but I think we should consider this and think about whether we can move back to that model ...

Reply:
Thanks for the feedback. I am raising an issue to keep track of this, happy to revisit the code to see if it can be brought back to a co-algebraic form